patches/tModLoader/Terraria/ModLoader/ModEmoteBubble.cs

```diff
@@ -0,0 +1,153 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Terraria.GameContent.UI;
+using Terraria.GameContent.UI.Elements;
+using Terraria.Localization;
+
+namespace Terraria.ModLoader
+{
+	public abstract class ModEmoteBubble : ModType<EmoteBubble, ModEmoteBubble>
+	{
+		/// <summary>
+		/// The file name of this emote's texture file in the mod loader's file space.
+		/// </summary>
+		public virtual string Texture => (GetType().Namespace + "." + Name).Replace('.', '/');
+
+		/// <summary>
+		/// The internal ID of this EmoteBubble.
+		/// </summary>
+		public int Type { get; internal set; }
+
+		/// <summary>
+		/// This is the <see cref="EmoteBubble"/> instance.
+		/// </summary>
+		public EmoteBubble EmoteBubble { get; internal set; }
```

Comment:
This should just be an accessor for the `Entity` property, see `ModItem` for an example

`public Item EmoteBubble => Entity;`